### PR TITLE
[MetricsAdvisor] Added missing DatasourceCredential types

### PR DESCRIPTION
**Pull request by kinelski**
Part of https://github.com/azure/azure-sdk-for-net/issues/18609.

Adding the three remaining `DatasourceCredential` types. It's pretty much the same thing as the `ServicePrincipalDatasourceCredential` we added in a previous PR.

### CODE REVIEW

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/Credential/ServicePrincipalDatasourceCredential.cs [hunk 1]
```diff
@@ -53,18 +53,6 @@ internal ServicePrincipalDatasourceCredential(DataSourceCredentialType dataSourc
         /// </summary>
         public string TenantId { get; set; }
 
-        /// <summary>
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/Credential/ServicePrincipalDatasourceCredential.cs:56]: Just removing the method to the end of the properties. Didn't want to mix them up.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/Models/DatasourceCredentialsTests.cs [hunk 2]
```diff
@@ -16,9 +17,14 @@ namespace Azure.AI.MetricsAdvisor.Tests
 {
     public class DatasourceCredentialsTests : MockClientTestBase
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/Models/DatasourceCredentialsTests.cs:18]: Non-live tests to make sure we're passing the secret to the service. The service never returns the secrets back to us, so there's no way to validate it live.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/api/Azure.AI.MetricsAdvisor.netstandard2.0.cs [hunk 3]
```diff
@@ -768,6 +768,11 @@ public abstract partial class DataFeedSource
         public static bool operator !=(Azure.AI.MetricsAdvisor.Models.DataFeedStatus left, Azure.AI.MetricsAdvisor.Models.DataFeedStatus right) { throw null; }
         public override string ToString() { throw null; }
     }
+    public partial class DataLakeGen2SharedKeyDatasourceCredential : Azure.AI.MetricsAdvisor.Models.DatasourceCredential
```
Comment (maririos) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/api/Azure.AI.MetricsAdvisor.netstandard2.0.cs:771]: I imagine there is a reason why the name needs to be super specific?

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/Credential/DataLakeGen2SharedKeyDatasourceCredential.cs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    [CodeGenModel("DataLakeGen2SharedKeyCredential")]
+    [CodeGenSuppress(nameof(DataLakeGen2SharedKeyDatasourceCredential), typeof(string), typeof(DataLakeGen2SharedKeyParam))]
+    public partial class DataLakeGen2SharedKeyDatasourceCredential
+    {
+        private string _accountKey;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DataLakeGen2SharedKeyDatasourceCredential"/> class.
+        /// </summary>
+        /// <param name="name">A custom unique name for this <see cref="SqlConnectionStringDatasourceCredential"/> to be displayed on the web portal.</param>
```
Comment (maririos) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/Credential/DataLakeGen2SharedKeyDatasourceCredential.cs:22]:
```suggestion
        /// <param name="name">A custom unique name for this <see cref="DataLakeGen2SharedKeyDatasourceCredential"/> to be displayed on the web portal.</param>
```

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/Credential/DataLakeGen2SharedKeyDatasourceCredential.cs [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        /// <param name="accountKey">The account key to be used for authentication.</param>
+        public DataLakeGen2SharedKeyDatasourceCredential(string name, string accountKey) : base(name)
+        {
+            Argument.AssertNotNullOrEmpty(accountKey, nameof(accountKey));
+
+            DataSourceCredentialType = DataSourceCredentialType.DataLakeGen2SharedKey;
+            AccountKey = accountKey;
+        }
+
+        internal DataLakeGen2SharedKeyDatasourceCredential(DataSourceCredentialType dataSourceCredentialType, string id, string name, string description, DataLakeGen2SharedKeyParam parameters) : base(dataSourceCredentialType, id, name, description)
```
Comment (maririos) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/Credential/DataLakeGen2SharedKeyDatasourceCredential.cs:32]: nit: could you have the `base` in the next line? easier to read

**Review by maririos — APPROVED** [review]

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/api/Azure.AI.MetricsAdvisor.netstandard2.0.cs (hunk 3 above)
Comment (kinelski) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/api/Azure.AI.MetricsAdvisor.netstandard2.0.cs:771]: Data Lake Gen 2 is the name of the service to which this credential is used. Shared Key specifies the type of credential. We will need to create other datasource credentials for different types of authentication.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/Credential/DataLakeGen2SharedKeyDatasourceCredential.cs (hunk 5 above)
Comment (kinelski) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/Credential/DataLakeGen2SharedKeyDatasourceCredential.cs:32]: Updated.

**Pull request merged**